content/hello-world-program.md

```diff
@@ -0,0 +1,295 @@
+- **Programs** on Solana are a particular type of account marked as executable and store instruction logic
+- **Transactions** on Solana are made up of instructions which are executed by programs
+
+# Overview
+
+Solana programs are written in the Rust programming language. Solana programs are stored in accounts with a unique address referred to as the program id. To understand programs on Solana we must first understand the basics of Solana’s Account Model.
+
+### Account Model
+
+All data stored on the Solana network are contained in what are referred to as accounts. Each account has its own unique address which is used to identify and access the account data. 
+
+There are two main categories of accounts:
+
+1. Data accounts which store state (ex. name, count) and must be owned by a program account
+2. Program accounts which store the executable programs (ex. instruction logic) and process instructions 
+
+There are two types of data accounts:
+
+1. Data accounts owned by the System Program (ex. user wallets)
+2. Data accounts owned by all other programs 
+
+There are two types of program accounts:
+
+1. Native program accounts which refer to Solana’s native programs (ex. System Program)
+2. All other program accounts 
+
+The diagram below demonstrates the relationship between program accounts and data accounts:
+
+![Screenshot Account Model Example Diagram](../assets/hello-world-account-diagram.png)
+
+Note that program ownership differs from a user which has authority over an account. For example, a user wallet with authority over a token account must sign a transaction in order to send tokens from the token account. The signed transaction then authorizes the token program, which owns the token account, to transfer tokens to another token account.
+
+See the image below for the full list of fields an account stores with its description.
+
+![Screenshot Account Fields](../assets/hello-world-account-fields.png)
+
+### Solana Programs
+
+To write Solana programs with Rust, we use the [solana_program](https://docs.rs/solana-program/latest/solana_program/index.html) library crate. Crates in Rust define functionality that can be shared with multiple projects. The `solana_program` crate acts as a standard library for Solana programs and contains modules and macros to help with the develop of our program. You can read more about Rust crates [here](https://doc.rust-lang.org/book/ch07-01-packages-and-crates.html).
+
+Paths are brought into scope with the [use](https://doc.rust-lang.org/stable/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html) keyword. In the example below, we bring into scope the [AccountInfo](https://docs.rs/solana-program/latest/solana_program/account_info/struct.AccountInfo.html) struct from the `account_info` module within the `solana_program` crate. You can read more about Rust modules [here](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html).
```

Comment:
@ZYJLiu "Paths" feels like it needs an introduction - what are paths?  Why do they need to be brought into scope?

Maybe something as short as the opening line of the previous doc page: https://doc.rust-lang.org/stable/book/ch07-03-paths-for-referring-to-an-item-in-the-module-tree.html
> To show Rust where to find an item in a module tree, we use a path in the same way we use a path when navigating a filesystem. If we want to call a function, we need to know its path.

e.g. **sugg:**
To "use" the modules available in the `solana_program` crate, we'll need to know how to invoke them by their "path" and bring it into the scope of our program. Paths are brought into scope...